engine/dlib/src/dmsdk/dlib/file_descriptor.h

```diff
@@ -0,0 +1,120 @@
+// Copyright 2021 The Defold Foundation
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#ifndef DMSDK_FILE_DESCRIPTOR_H
+#define DMSDK_FILE_DESCRIPTOR_H
+
+#include <dmsdk/dlib/array.h>
+#if defined(_WIN32)
+#include <winsock2.h>
+#else
+#include <poll.h>
+#endif
+
+/*# SDK File Descriptor API documentation
+ * File Descriptor functions.
+ *
+ * @document
+ * @name FileDescriptor
+ * @namespace dmFileDescriptor
+ * @path engine/dlib/src/dmsdk/dlib/file_descriptor.h
+ */
+namespace dmFileDescriptor
+{
+    /*#
+     * Poll events
+     * @enum
+     * @name PollEvent
+     * @member EVENT_READ
+     * @member EVENT_WRITE
+     * @member EVENT_ERROR
+     */
+    enum PollEvent
+    {
+        EVENT_READ   = 0,
+        EVENT_WRITE  = 1,
+        EVENT_ERROR  = 2,
+    };
+
+    /*#
+     * Poller
+     * @struct
+     * @name Poller
+     */
+    struct Poller
+    {
+        #if defined(_WIN32)
+        dmArray<WSAPOLLFD> m_Pollfds;
+        #else
+        dmArray<pollfd> m_Pollfds;
+        #endif
+    };
```

Comment:
This makes it really hard to make it platform independent for our platforms.
Use a typedef instead.

Reply:
Yep, added!